src/main/java/org/terasology/launcher/packages/RepositoryHandler.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.launcher.packages;
+
+import java.util.List;
+
+/**
+ * Provides method to fetch all packages from a repository.
+ */
+interface RepositoryHandler {
```

Comment:
Should this be a [`@FunctionalInterface`](https://docs.oracle.com/javase/8/docs/api/java/lang/FunctionalInterface.html)?

Reply:
Sometimes I really forget that I'm on Java 8 :sweat_smile: